src/Waives.Http/HttpRequestMessageExtensions.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System.IO;
+using System.Net.Http;
+using System.Threading.Tasks;
+
+namespace Waives.Http
+{
+    public static class HttpRequestMessageExtensions
```

Comment:
✋ This should be internal, along with `HttpRequestMessage.CloneAsync()`. 

Side note: I'm not massively keen on this approach, to be honest; constructing the message from parameters as mentioned in the PR description would have been a much cleaner way to achieve the same thing. This appears to introduce a lot of code both here and in the tests which is ancillary at best to achieving the aim of "sending the same message twice".